### DISCUSSION TITLE
How can i find a repository is on user/personal account or on a github organization

### DISCUSSION THREAD

**Discussion by githubofkrishnadhas** [community/community#130988] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hello,

I was wondering if you can help me with finingd a repository is on user/personal account or on a github organization. i have already checked docs cant find anything. i also cehcked the available github envrionment variables for this. still no luck. 

Any help is really appreciated.

**Comment by githubofkrishnadhas** [comment]
Hey @lilly452 
What i would relaly want to know is if we can programatically get the details. it was to automate something.

**Reply by Daeraxa** [reply, marked as answer]
You could use the GH CLI?
For example `pulsar` is a repo within the `pulsar-edit` org but my fork is not, it is on my personal account: 

```sh
$ gh repo view pulsar-edit/pulsar --json owner,name,isInOrganization
{
  "isInOrganization": true,
  "name": "pulsar",
  "owner": {
    "id": "O_kgDOBna96A",
    "login": "pulsar-edit"
  }
}

$ gh repo view daeraxa/pulsar --json owner,name,isInOrganization
{
  "isInOrganization": false,
  "name": "pulsar",
  "owner": {
    "id": "MDQ6VXNlcjU4MDc0NTg2",
    "login": "Daeraxa"
  }
}
```

It is probably available on the normal GH API as well but this was just an easy way of doing it.

**Reply by githubofkrishnadhas** [reply]
Thanks. This is awesome

**Comment by githubofkrishnadhas** [comment]
Wow thanks. I was using GitHub rest api for the automation purpose and this gave me a hint to find the answer i was looking for. The answer was simple which is why i missed it.

To use Github ret api,

* use repo end point which is     `url = f"https://api.github.com/repos/{GITHUB_REPOSITORY}"` where GITHUB_REPOSITORY is the full name of repo including org name/reponame or username/reponame.

we can get the response back and sort for     `owner_type = repo_info['owner']['type']`

using python will be something similar to

```
import requests
import os

def get_repo_owner_type():
    url = f"https://api.github.com/repos/{GITHUB_REPOSITORY}"
    headers = {
            "Accept": "application/vnd.github+json",
            "Authorization": f"Bearer {os.getenv('GH_TOKEN')}",
            "X-GitHub-Api-Version": "2022-11-28"
        }
    response = requests.get(url, headers=headers)

    if response.status_code == 200:
        repo_info = response.json()
        owner_type = repo_info['owner']['type']
        return owner_type
    else:
        print(f"Error: {response.status_code} - {response.json()}")
        return None

# Example usage

owner_type = get_repo_owner_type()
if owner_type:
    print(f"The repository '{GITHUB_REPOSITORY}' is owned by a {owner_type}.")

```

Rememeber to pass proper authentication token for successful api call.

**Comment by queenofcorgis** [comment]
We’re glad you could get this solved and we want to make sure others can do the same so we’ve reopened this discussion for better discoverability for our members.

**Reply by githubofkrishnadhas** [reply]
Wow. Thats great. But still this is not completely feasible  by using GitHub rest API i guess.

The above github rest api end point for repos  `https://api.github.com/repos/repo_name` can only be used  for public repos or private repos i own.

![image](https://github.com/user-attachments/assets/94abd4d3-2a6f-464c-b96f-b89a1b8d56b1)

[get-a-repository](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository)

lets say i want to determine a repo that doesn't own by me or i don't know whether it's a personal repo or a org one, i can't determine using above method as i need a form of auth to determine so.

It could be helpful if we can introduce something for this.   

use case
---------

lets say i am writing a github automation or action which i wanna later release to github market place. it invlves some complex logic inside to determine if a repo belongs to a user or a org and respectively use api end point for org or user respectively, this could be very helpful
